### PR TITLE
Handle point special cases.

### LINKED ISSUE #9: Finishing games.
When the conditions are met for a winner, set the `winner` and `finished_at` to now.

### PR DESCRIPTION
**Pull request by andrewhood125**
Fixes #9, #10 
  - [ ] Spec

### CODE REVIEW

File: app/models/game.rb [hunk 1]
```diff
@@ -20,6 +20,32 @@ def score
...
+        self.server_id = self.challenger_id
+      end
+    end
+    # check if game has ended
+    # won by at least 2 points?
+    if (score[:challenger] - score[:opponent]).abs > 1
+      leader, value = score.max_by { |k,v| v }
+      if value >= 10
+        self.finished_at = DateTime.now
+        self.winner_id = send(leader).id
```
Comment (forkata) [inline app/models/game.rb]: A bit confused why you are using `send` here?

File: app/models/game.rb [hunk 2]
```diff
@@ -20,6 +20,32 @@ def score
...
+    super(options.merge({
+      include: [
+        { challenger: { except: [:api_token] }},
+        { winner: { except: [:api_token] }},
+        { opponent: { except: [:api_token] }},
+        { server: { except: [:api_token] }},
+      ], methods: [ :score ] }))
+  end
+
+  def scored
```
Comment (forkata) [inline app/models/game.rb]: It looks like this is responsible for updating the server and also finalizing the game. Might be good to break those out into two separate methods. That would also make it easier to test :)

Comment (andrewhood125) [inline app/models/game.rb]: Good point! I'm unhappy with how all the processing is done for these special cases. This blocks the `create` action on points. I wanted to get logic down but before this gets to master I want to refactor it. I'm leaning towards emitting an event `scored` and then subscribe to that event for checking if the game is finished and alternating service as well.

File: app/models/game.rb (hunk 1 above)
Comment (andrewhood125) [inline app/models/game.rb]: Game has the relationships `challenger` and `opponent`. `leader` is a symbol `:challenger` or `:opponent` of the player with the most points in the game. So what we're doing here is calling the relationship based on the on the symbol. 

I don't love how this is done, it's limited based on the hash you get back from `score`. I might refactor so I can avoid the extra queries and `send`.

**Pull request merged**